### PR TITLE
[WIP] GRPC framework for out-tree implementation of cloud providers

### PR DESCRIPTION
**Pull request by ggaurav10**

### CODE REVIEW

**Comment by hardikdr** [comment]
@ggaurav10 @amshuman-kr Thanks for the PR.
Interesting feature, can we also document it around how someone can implement the out-of-tree driver. Eventually we would like to move all of the cloud-specific code out of the repo.

**Comment by hardikdr** [comment]
@ggaurav10 can you please provide some more information around how are you trying to add the feature exactly - may be either via PR description or proposal?
It's really hard to review it otherwise for anyone from outside.

File: cmd/external-aws-client/external-aws-client.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package main
```
[the remark this answers is not in the document]
Comment (hardikdr) [reply cmd/external-aws-client/external-aws-client.go]: Thanks for drawing the attention Matthias, I think it has been fixed by Gaurav now.

[the remark this answers is not in the document]
Comment (ggaurav10) [reply cmd/external-aws-client/external-aws-client.go]: done

**Pull request closed without merging**